examples/with-cloudinary/pages/index.tsx

```diff
@@ -0,0 +1,176 @@
+              href="https://vercel.com/new/clone?repository-url=https://github.com/vercel/next.js/tree/canary/examples/with-cloudinary&project-name=nextjs-image-gallery&repository-name=with-cloudinary&env=NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME,CLOUDINARY_API_KEY,CLOUDINARY_API_SECRET,CLOUDINARY_FOLDER&envDescription=API%20Keys%20from%20Cloudinary%20needed%20to%20run%20this%20application"
+              target="_blank"
+              rel="noreferrer"
+            >
+              Clone and Deploy
+            </a>
+          </div>
+          {images.map(({ id, public_id, format, blurDataUrl }) => (
+            <Link
+              key={id}
+              href={`/?photoId=${id}`}
+              as={`/p/${id}`}
+              id={`photo-${id}`}
+              shallow
+              className="after:content group relative cursor-zoom-in after:pointer-events-none after:absolute after:inset-0 after:rounded-lg after:shadow-highlight"
+            >
+              <Image
+                alt="Next.js Conf photo"
+                className="transform rounded-lg brightness-90 transition will-change-auto group-hover:brightness-110"
+                style={{ transform: 'translate3d(0, 0, 0)' }}
+                placeholder="blur"
+                blurDataURL={blurDataUrl}
+                src={`https://res.cloudinary.com/${process.env.NEXT_PUBLIC_CLOUDINARY_CLOUD_NAME}/image/upload/c_scale,w_720/${public_id}.${format}`}
+                width={720}
+                height={480}
+                sizes="(max-width: 640px) 100vw,
+                  (max-width: 1280px) 50vw,
+                  (max-width: 1536px) 33vw,
+                  25vw"
+              />
+            </Link>
+          ))}
+        </div>
+      </main>
+      <footer className="p-6 text-center text-white/80 sm:p-12">
+        Thank you to{' '}
+        <a
+          href="https://edelsonphotography.com/"
+          target="_blank"
+          className="font-semibold hover:text-white"
+          rel="noreferrer"
+        >
+          Josh Edelson
+        </a>
+        ,{' '}
+        <a
+          href="https://www.newrevmedia.com/"
+          target="_blank"
+          className="font-semibold hover:text-white"
+          rel="noreferrer"
+        >
+          Jenny Morgan
+        </a>
+        , and{' '}
+        <a
+          href="https://www.garysextonphotography.com/"
+          target="_blank"
+          className="font-semibold hover:text-white"
+          rel="noreferrer"
+        >
+          Gary Sexton
+        </a>{' '}
+        for the pictures.
+      </footer>
+    </>
+  )
+}
+
+export default Home
+
+export async function getStaticProps() {
+  const results = await cloudinary.v2.search
+    .expression(`folder:${process.env.CLOUDINARY_FOLDER}/*`)
+    .sort_by('public_id', 'desc')
+    .max_results(400)
```

Comment:
Planning on adding infinite scrolling + smoothing out some animations next week, but for now, I'd love to ship it since we're sending out an email tomorrow morning to all Next.js Conf attendees with the photo gallery. Lemme know if that sounds alright to you.

For context, we've kept folks waiting for too long since I was originally planning to send this out early last week and I've already told a couple folks that we were sending it tomorrow.

Reply:
Sounds good to do in a follow-up 👍